dash_agent/README.md

```diff
@@ -1,39 +1,174 @@
 
-TODO: List what your package can do. Maybe include images, gifs, or videos.
+```shell
+dart pub global activate dash-cli
+```
 
-## Getting started
+Now, run the below shared command in the terminal once again:
 
-TODO: List prerequisites and provide or point to information on how to
-start using the package.
+```shell
+dash-cli create {{agent}}
+```
 
 ## Usage
 
-TODO: Include short and useful examples for package users. Add longer examples
-to `/example` folder.
+This package contain the building blocks for creating dash agents:
+
+### AgentConfiguration
+
+The main part of the framework that glues together your agent configuration is `AgentConfiguration`. It is the base class for creating agent.
+
+Sample example of `AgentConfiguration` for demonstration purpose:
+
+```dart
+class MyAgent extends AgentConfiguration {
+  final docsSource = DocsDataSource();
+
+  @override
+  List<DataSource> get registeredDataSources => [docsSource];
+
+  @override
+  List<Command> get registerSupportedCommands =>
+      [AskCommand(docsSource: docsSource)];
+}
+```
+
+The above sample AgentConfiguration object taking data sources and supported commands. Both of them are explained below:
+
+- `DataSource`: Data sources will enable you to pass on any form of data that your agent might need to perform its intended tasks.
+- `Command`: Commands are the specialised tasks you want your agents to perform (like understanding the developer query, refactoring, code generation, etc).
+
+### Datasource
+
+As described above data sources lets you attach the data that your agent will need to perform its tasks. It can be anything from raw texts, json, files data, or even webpages.
+
+Sample example for DataSource for demostration purpose:
+
+```dart
+class DocsDataSource extends DataSource {
+
+ /// Enables you to pass data stored in files and directories in you local system.
+ @override
+ List<FileDataObject> get fileObjects => [
+       FileDataObject.fromFile(File(
+           'your_file_path')),
+        FileDataObject.fromDirectory(Directory(
+            'directory_path_to_data_source'))
+     ];
+ 
+
+ /// Enables you to pass in raw string and json data
+ @override
+ List<ProjectDataObject> get projectObjects =>
+     [ProjectDataObject.fromText('Data in form of raw text')];
+
+
+ /// Enables you to pass web data your agent by passing web pages urls or sitemaps in the object.
+  @override
+ List<WebDataObject> get webObjects =>
+     [WebDataObject.fromWebPage('https://sampleurl.com'), 
+     WebDataObject.fromSiteMap('https://sampleurl.com/sitemap.xml')];
+}
+```
+
+To create any data object, the easiest (and also the recommended way) is to call static functions of the above shared bases class of the object. For example if you want to store the json data. You can add the json data as shared below:
 
 ```dart
-const like = 'sample';
+final yourJson = {'key': 'data'};
+final jsonDataSource = ProjectDataSource.fromJson(yourJson);
 ```
```

Comment:
This is better mentioned in the documentation. The above mentioned examples are suffice.